database/CoreDatabase.java

```diff
@@ -731,12 +731,15 @@ private void countFn() {
                 } catch (TypeDBException e) {
                     if (e.code().isPresent() && e.code().get().equals(DATABASE_CLOSED.code())) {
                         break;
-                    } else if (e.code().isPresent() && (e.code().get().equals(TRANSACTION_CONSISTENCY_MODIFY_DELETE_VIOLATION.code()) ||
+                    } else if (e.code().isPresent() && (
+                            e.code().get().equals(TRANSACTION_CONSISTENCY_MODIFY_DELETE_VIOLATION.code()) ||
                             e.code().get().equals(TRANSACTION_CONSISTENCY_DELETE_MODIFY_VIOLATION.code()) ||
-                            e.code().get().equals(TRANSACTION_CONSISTENCY_EXCLUSIVE_CREATE_VIOLATION.code()))) {
+                            e.code().get().equals(TRANSACTION_CONSISTENCY_EXCLUSIVE_CREATE_VIOLATION.code())
+                    )) {
                         countJobNotifications.release();
                     } else {
-                        throw e;
+                        LOG.error("Background statistics counting received exception.", e);
+                        countJobNotifications.release();
```

Comment:
we want to log unexpected errors when counting statistics (such as maybe schema lock can't be acquired), but we generally want to continue counting...

The risk is that we keep counting! However, the thread should be stopped when the database is closed/deleted, so this is safe.